lab5_BankAccount.py

```diff
@@ -0,0 +1,89 @@
+class BankAccount:
+
+    __current_id = 0
+
+    def __new__(cls, *args, **kwargs):
+        cls.__current_id += 1
+        return super().__new__(cls)
+
+    def __init__(self, owner_name, balance = 0.0):
+        self.__id = self.check_id(self.get_id())
+        self.__owner_name = self.check_owner_name(owner_name)
+        self.__balance = self.check_money((balance))
+
+    def __str__(self):
+        return f'Account {self.__id}: Owner name - {self.__owner_name}, Balance: {self.__balance}'
+
+    @classmethod
+    def get_id(cls):
+        return cls.__current_id
+
+    def deposit(self, money):
+        self.__balance += self.check_money(money)
+        print(f'You have successfully topped up your account by {money}')
+
+    def withdraw(self, money):
+        if money > self.__balance:
+            raise Exception("you don't have that much money")
+        self.__balance -= self.check_money(money)
+        print(f'you have successfully withdrawn {money} from your account')
+
+    # @property
+    # def id(self):
+    #     return self.__id
+    #
+    # @id.setter
+    # def id(self, new_id):
+    #     self.__id = self.check_id(new_id)
+
+    @property
+    def owner_name(self):
+        return self.__owner_name
+
+    @owner_name.setter
+    def owner_name(self, new_name):
+        self.__owner_name = self.check_owner_name(new_name)
+
+    @property
+    def balance(self):
+        return self.__balance
+
+    @balance.setter
+    def balance(self, new_balance):
+        self.__balance = self.check_money(new_balance)
+
+    @staticmethod
```

Comment:
remove static method and create such function in main file

Reply:
These methods are used to check that the input data is of the correct type. They should be in the class "BankAccount" as staticmethods because they will not be used anywhere else